lnrpc/invoicesrpc/invoices_server.go

```diff
@@ -446,3 +464,27 @@ func (s *Server) LookupInvoiceV2(ctx context.Context,
 
 	return CreateRPCInvoice(&invoice, s.cfg.ChainParams)
 }
+
+// HtlcModifier is a bidirectional streaming RPC that allows a client to
+// intercept and modify the HTLCs that attempt to settle the given invoice. The
+// server will send HTLCs of invoices to the client and the client can modify
+// some aspects of the HTLC in order to pass the invoice acceptance tests.
+func (s *Server) HtlcModifier(
+	modifierServer Invoices_HtlcModifierServer) error {
+
+	// Ensure that there is only one invoice HTLC modifier RPC server
+	// instance.
+	if !s.invoiceHtlcModifierActive.CompareAndSwap(0, 1) {
+		return ErrInvoiceHtlcModifierAlreadyExists
+	}
+	defer s.invoiceHtlcModifierActive.CompareAndSwap(1, 0)
```

Comment:
follow on from previous comment: i think this is basically redundant since we can just check in the modifier if `callback` is nil or not & ensure that if it is not nil, then no new interceptor can be registered